normandy/recipes/models.py

```diff
@@ -174,3 +183,61 @@ def country(self):
     @cached_property
     def request_time(self):
         return self.request.received_at
+
+
+class ApprovalRequest(models.Model):
+    recipe = models.ForeignKey(Recipe, related_name='approval_requests')
+    created = models.DateTimeField(default=datetime.now)
+    creator = models.ForeignKey(User)
+    active = models.BooleanField(default=True)
+
+    class IsNotActive(Exception):
+        pass
+
+    @property
+    def is_approved(self):
+        return self.approval is not None
+
+    def approve(self, user):
+        if self.active:
+            approval = Approval(approval_request=self, creator=user)
+            approval.save()
+
+            self.active = False
+            self.save()
+
+            self.recipe.approval = approval
+            self.recipe.save()
+        else:
+            raise self.IsNotActive('Approval request has already been closed.')
```

Comment:
I really like this pattern of raising an exception from the class itself, it's neato!